File: examples/simple_form.py
"""
Simple Formora Example

This example demonstrates basic form creation with Formora.
"""

from formora import Form, CssFramework, Rule

# Create a simple contact form
html = (
    Form("contact")
    .title("Contact Us")
    .description("We'd love to hear from you!")
    .css(CssFramework.BOOTSTRAP)
    .text("name", "Your Name", required=True, rules=[Rule.required(), Rule.min_length(2)])
    .email("email", "Email Address", required=True, rules=[Rule.required(), Rule.email()])
    .select("topic", "Topic", options=[
        ("General Inquiry", "general"),
        ("Support", "support"),
        ("Sales", "sales")
    ])
    .textarea("message", "Message", rows=5, required=True)
    .submit_label("Send Message")
    .success_message("Thanks for reaching out! We'll get back to you soon.")
    .build()
)

print(html)

File: formora-js/examples/simple_form.ts
/**
 * Simple contact form example.
 * TypeScript equivalent of examples/simple_form.py
 */

import init, { Form, Rule, CssFramework } from "../wasm/formora_js";

async function main() {
  await init();

  const html = new Form("contact-form")
    .title("Contact Us")
    .description("Send us a message and we'll get back to you.")
    .css(CssFramework.bootstrap())
    .text("name", "Full Name", "Your full name", true, undefined, undefined, [
      Rule.required("Name is required"),
      Rule.minLength(2, "Name must be at least 2 characters"),
    ])
    .email("email", "Email Address", true, undefined, undefined, [
      Rule.required("Email is required"),
      Rule.email("Please enter a valid email"),
    ])
    .select(
      "subject",
      "Subject",
      [
        ["General Inquiry", "general"],
        ["Support", "support"],
        ["Billing", "billing"],
        ["Other", "other"],
      ],
      true
    )
    .textarea(
      "message",
      "Message",
      5,
      "Type your message here...",
      true,
      undefined,
      undefined,
      [
        Rule.required("Message is required"),
        Rule.minLength(10, "Message must be at least 10 characters"),
      ]
    )
    .submitLabel("Send Message")
    .successMessage("Thank you! We'll be in touch soon.")
    .build();

  console.log(html);
}

main().catch(console.error);
